pkg/k8s/fake/secret.go

```diff
@@ -4,15 +4,12 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
-// SecretKey is a fake key
-const SecretKey string = "Key"
-
-// SecretValue is a fake value
-const SecretValue string = "Value"
-
 // Secret creates a fake secret with defined name
 func Secret(name string, namespace string) *v1.Secret {
-	data := make(map[string]string)
-	data[SecretKey] = SecretValue
-	return &v1.Secret{ObjectMeta: ObjectMeta(name, namespace), StringData: data}
+	return &v1.Secret{ObjectMeta: ObjectMeta(name, namespace)}
+}
+
+// SecretWithType creates a fake secret with defined name
+func SecretWithType(name string, namespace string, secretType v1.SecretType) *v1.Secret {
+	return &v1.Secret{ObjectMeta: ObjectMeta(name, namespace), Type: secretType}
 }
```

Comment:
For some tests we don't need a type

Reply:
Yes, but by definition every K8s secret has a type. Therefore a constructor function should take a type as argument. Tests not interested in a particular type should just choose one, e.g. `Opaque`. You may also allow specifying the empty string in case tests must deal with the situation that the secret type is not set in a secret struct.

Reply:
See 031bcb7a937eea2b3a74ac5184b59babe23630af